### DISCUSSION TITLE
Help, why hasn't pnpm installed the dependencies of "@tarojs/react"?

### DISCUSSION THREAD

**Discussion by zhaojinzhao** [pnpm/pnpm#8894] [Q&A] [answered]
In my project, i installed "@tarojs/react" library with pnpm, but the dependencies of "@tarojs/react" didn't be installed. there is no node_modules dir in @tarojs/react dir. why?

**My project structure**
![image](https://github.com/user-attachments/assets/ed572e1e-2262-43e6-8eef-2e2c33e80029)

**@tarojs/react's dependencies**
![image](https://github.com/user-attachments/assets/6cc1c0bc-a8bf-4375-bdaa-c332efdd296b)

**Comment by zkochan** [comment, marked as answer]
The dependencies of `@tarojs/react` are inside `node_modules/.pnpm`. Read this article to understand how it works: https://pnpm.io/blog/2020/05/27/flat-node-modules-is-not-the-only-way

**Reply by zhaojinzhao** [reply]
Got it. thank you.
